Bottom: 727cee94bab8bce369b3175d6a826e53a02417ee
Top:    dd8d933f069e480214d8c94c47c8e63a86f99fa3
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-05-13 08:31:35 +0800

Refresh of vhost_net-passing-raw-xdp-buff

---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index cbb4634..8572960 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -628,6 +628,12 @@ static void handle_tx_copy(struct vhost_net *net)
 		vq->heads[nheads].id = cpu_to_vhost32(vq, head);
 		vq->heads[nheads].len = 0;
 
+		err = vhost_net_build_xdp(nvq, &msg.msg_iter, &xdp);
+		if (!err)
+			msg.msg_control = &xdp;
+		else
+			msg.msg_control = NULL;
+
 		total_len += len;
 		if (total_len < VHOST_NET_WEIGHT &&
 		    vhost_has_more_pkts(net, vq)) {
@@ -752,14 +758,7 @@ static void handle_tx_zerocopy(struct vhost_net *net)
 			msg.msg_control = NULL;
 			ubufs = NULL;
 		}
-<<<<<<< HEAD
-=======
 
-		err = vhost_net_build_xdp(nvq, &msg.msg_iter, &xdp);
-		if (!err) {
-			msg.msg_control = &xdp;
-		}
->>>>>>> 185fce3... vhost_net: passing raw xdp buff to tun
 		total_len += len;
 		if (total_len < VHOST_NET_WEIGHT &&
 		    vhost_has_more_pkts(net, vq)) {
